Graphics/Implicit/ObjectUtil/GetBoxShared.hs

```diff
@@ -0,0 +1,164 @@
+-- Copyright 2014 2015 2016, Julia Longtin (julial@turinglace.com)
+-- Copyright 2015 2016, Mike MacHenry (mike.machenry@gmail.com)
+-- Implicit CAD. Copyright (C) 2011, Christopher Olah (chris@colah.ca)
+-- Released under the GNU AGPLV3+, see LICENSE
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeSynonymInstances   #-}
+
+module Graphics.Implicit.ObjectUtil.GetBoxShared where
+
+import Prelude ((==), max, min, foldr, (/), ($), fmap, (.), not, filter, foldMap, Fractional, Bool, Eq)
+import {-# SOURCE #-} Graphics.Implicit.Primitives
+    ( Object(getBox) )
+import Graphics.Implicit.Definitions
+    ( SharedObj(..), ComponentWiseMultable((⋯*)), ℝ3, ℝ2, ℝ )
+import Data.VectorSpace
+    ( AdditiveGroup((^+^), zeroV, (^-^)),
+      InnerSpace,
+      VectorSpace(Scalar) )
+import Graphics.Implicit.MathUtil (infty,  reflect )
+
+
+------------------------------------------------------------------------------
+-- | Ad-hoc methods we need to share code between 2D and 3D. With the exception
+-- of 'corners', these are actually all standard methods of other classes,
+-- which we don't have access to due to the choice representation for R2 and
+-- R3.
+--
+-- This class is unnecessary if we were to implement #283.
+class VectorStuff vec where
+  -- | Equivalent to 'Prelude.pure'
+  uniformV :: ℝ -> vec
+  -- | Equivalent to 'Control.Applicative.liftA2'
+  pointwise :: (ℝ -> ℝ -> ℝ) -> vec -> vec -> vec
+  -- | Equivalent to 'Data.Foldable.toList'
+  elements :: vec -> [ℝ]
+  -- | Given a bounding box, produce the points at each corner.
+  corners :: (vec, vec) -> [vec]
+
+
+instance VectorStuff ℝ2 where
+  uniformV x = (x, x)
+  corners (p1@(x1, y1), p2@(x2, y2)) =
+    [ p1
+    , (x1, y2)
+    , (x2, y1)
+    , p2
+    ]
+  pointwise f (x1, y1) (x2, y2) = (f x1 x2, f y1 y2)
+  elements (x, y) = [x, y]
+  {-# INLINABLE uniformV #-}
```

Comment:
no inlinable for corners?

Reply:
I don't expect it to matter, but happy to add them!